### PR TITLE
KIP-0030: SpireKey dApp <> Wallet Connection

### PR DESCRIPTION
**Pull request by EnoF**
Scoped KIP down to dapp <> wallet connection

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208668844268904

### CODE REVIEW

**Review by Randynamic — DISMISSED** [review]

**Review by Randynamic — APPROVED** [review]

File: kip-0030.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+>;
+export type Account = {
+  alias: string;
+  accountName: string;
+  minApprovals: number;
+  minRegistrationApprovals: number;
+  balance: string;
+  devices: Device[];
+  guard?: Guard;
+  keyset?: Keyset;
```
Comment (emilypi) [inline kip-0030.md:193]: Is this guard significantly different from the keyset? It would be good to explain the difference.

Comment (EnoF) [reply kip-0030.md:193]: I'll update the supporting text around it. :+1:

**Comment by alber70g** [comment]
As per discussed with @Randynamic and @EnoF I propose to combine this with KIP-0017 and create a new KIP as an extension. I even think we can use this more widely. 

I would propose to make a new KIP describing **Spirekey Connect**. 

**Spirekey Connect** is a comprehensive messaging interface that can then be used on various transport layers, like WalletConnect, URL-#-based (eg `wallet.kadena.io/spirekey?returnURL=mydapp.com/spirekey#<message>`), endpoint based `POST <host>/spirekey`, `window.spirekey` or `window.kadena.spirekey(message)`.

That way we can separate the transport layer from the messaging layer and generalize the way dapps communicate with Kadena wallets.

The messages give support for:
* `connect`: open the wallet and return keys (and accounts?)
* `getAccounts`: retrieve accounts relevant to a certain key
* `signWithRequirements`: a message that contains a finished transaction (similar to Chainweaver v2s /quicksign) and the requirements that the wallet needs to fulfill to complete that transaction (like moving fungibles across chains, etc)
* `transactionRequest`: a message that contains a piece of code with an incomplete transaction (similar to the current /sign of chainweaver v2) . The code would have `read-x` statements that, based on some additional metadata, can be filled by the wallet by presenting a form based on the metadata.

Note: `connect` and `getAccounts` are not a hard requirement. A dApp can also request the account from the user from within the dapp, and submit a `signWithRequirements` or `transactionRequest` afterwards.

**Comment by alber70g** [comment]
Next to this, we should understand what we mean with SpireKey. What is it? There's currently too much noise:  
Is it a wallet, signing using WebAuthn, using a smart contract to make smart wallets governed by a WebAuthn key, is it r-accounts?

**Pull request closed without merging**